extensions/mochi-gg/src/market.tsx

```diff
@@ -0,0 +1,416 @@
+        return;
+      }
+
+      try {
+        const query = search.toLowerCase().replace(/ /g, "-");
+        if (query.length < 2) {
+          return;
+        }
+
+        setBgSearch(true);
+        const {
+          data: { data },
+        } = await axios.get<ICoinCompareResult>(
+          `https://api.mochi.pod.town/api/v1/defi/coins/compare?base=${query}&interval=7&target=usd`
+        );
+        if (data.base_coin_suggestions) {
+          mapLimit<ICoinSuggestion, IMarketData>(
+            data.base_coin_suggestions,
+            5,
+            async (item: ICoinSuggestion) => {
+              console.log("[getCoins]", JSON.stringify(item, null, 4));
+              return getCoinDetails(item.id);
+            },
+            (err, results) => {
+              console.error("[getCoins]", search, err);
+              const items = _.filter(results, (item: IMarketData) => !!tokenLookUpNoExtra[item?.id]);
+              setExtraData(items as IMarketData[]);
+            }
+          );
+        } else {
+          const dt = await getCoinDetails(query);
+          if (!tokenLookUpNoExtra[dt.id]) {
+            setExtraData([dt]);
+            return;
+          }
+        }
+      } catch (err) {
+        console.error(`[ERROR.getCoins]`, err);
+      } finally {
+        setBgSearch(false);
+      }
+    }
+
+    getCoins(searchText);
+  }, [tokenLookUpNoExtra, searchText, canDoBgSearch]);
+
+  const fues = useMemo(() => {
+    const options = {
+      includeScore: false,
+      // Search in `author` and in `tags` array
+      keys: ["name", "symbol"],
+    };
+    const src = sortBy
+      ? _.chain(data?.data).concat(extraData).uniqBy("id").sortBy(sortBy).reverse().value()
+      : _.chain(data?.data).concat(extraData).unionBy("id").value();
+    return new Fuse(src, options);
+  }, [data, extraData, sortBy]);
+
+  const filteredList = useMemo(() => {
+    if (!searchText) {
+      const src = sortBy ? _.chain(data?.data).sortBy(sortBy).uniqBy("id").reverse().value() : data?.data || [];
+      return src.map((item) => ({ item })) || [];
+    }
+
+    return fues.search(searchText);
+  }, [fues, searchText, sortBy, data]);
+
+  useEffect(() => {
+    if (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong",
+        message: error.message,
+      });
+    }
+  }, [error]);
+
+  const AddWatchListAction = (item: ITickerBaseCoin) =>
+    user?.id ? (
+      <Action
+        title={watchingMap[item.id] ? "Remove From Watch List" : "Add to Watch List"}
+        onAction={async () => {
+          if (watchingMap[item.id]) {
+            await removeTokenFromWatchlist(item);
+          } else {
+            await addTokenToWatchlist(item);
+            revalidate();
+          }
+        }}
+        // @ts-ignore
+        style={watchingMap[item.id] ? "destructive" : "regular"}
+        icon={watchingMap[item.id] ? Icon.Trash : Icon.Plus}
```

Comment:
Not sure the `@ts-ignore` is useful here. Why not early returning if there's no corresponding ID in the `watchingMap` object?